internal/rpc/httpproxy_test.go

```diff
@@ -0,0 +1,97 @@
+// Copyright 2024 Canonical.
+package rpc_test
+
+import (
+	"context"
+	"encoding/pem"
+	"net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+
+	qt "github.com/frankban/quicktest"
+	"github.com/juju/juju/core/network"
+	jujuparams "github.com/juju/juju/rpc/params"
+
+	"github.com/canonical/jimm/v3/internal/dbmodel"
+	"github.com/canonical/jimm/v3/internal/rpc"
+)
+
+func TestProxyHTTP(t *testing.T) {
+	c := qt.New(t)
+	ctx := context.Background()
+	// we expect the controller to respond with TLS
+	fakeController := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		_, err := w.Write([]byte("OK"))
+		c.Assert(err, qt.IsNil)
```

Comment:
I'd like to see one test where the controller doesn't return an HTTP 200 and the client gets that back.

Reply:
added